### DISCUSSION TITLE
No options to use it with Open Sourced models?

### DISCUSSION THREAD

**Discussion by venturaEffect** [InternLM/MindSearch#272] [Ideas]
The project is awesome. But I think it fails in not providing an option to use Open Source LLMs.

The implementation if doing so, would be massive.

But, maybe I'm wrong and haven't seen that it is possible.

Anyway, congrats on the project!

**Comment by idontwantagirlfriend** [comment]
It should be possible to run your own llm backend and use any OpenAI compatible endpoint. Probably tweak with mindsearch/agent/models.py, and set your backend url at backend.py. Try it out!
